front/lib/api/assistant/pubsub.ts

```diff
@@ -96,7 +97,7 @@ async function handleUserMessageEvents(
 ): Promise<Result<UserMessageType, PubSubError>> {
   const promise: Promise<Result<UserMessageType, PubSubError>> = new Promise(
     (resolve) => {
-      void (async () => {
+      void wakeLock(async () => {
```

Comment:
You're likely missing the `retryAgentMessageWithPubSub` one?

Reply:
Yep and the cancel I guess ?

Reply:
Otherwise LGTM!

Reply:
Well cancel is not waiting on anything